ci/test-vmc.sh

```diff
@@ -0,0 +1,421 @@
+        capi_pod_name=$(echo ${capi_pod} | cut -d'/' -f 2)
+        kubectl logs ${capi_pod_name} -c manager -n capi-system > ${CLUSTER_LOG_DIR}/capi/${capi_pod_name} || true
+    done
+
+    echo "=== Saving capv logs ==="
+    mkdir -p ${CLUSTER_LOG_DIR}/capv
+    kubectl get -n capv-system pods -o name | awk '{print $1}' | while read capv_pod; do
+        capv_pod_name=$(echo ${capv_pod} | cut -d'/' -f 2)
+        kubectl logs ${capv_pod_name} -c manager -n capv-system > ${CLUSTER_LOG_DIR}/capv/${capv_pod_name} || true
+    done
+
+    echo "=== Saving cluster_api.yaml ==="
+    mkdir -p ${CLUSTER_LOG_DIR}/cluster_api
+    kubectl get cluster-api -A -o yaml > ${CLUSTER_LOG_DIR}/cluster_api/cluster_api.yaml || true
+}
+
+function setup_cluster() {
+    export KUBECONFIG=$KUBECONFIG_PATH
+    rm -rf ${GIT_CHECKOUT_DIR}/jenkins || true
+
+    echo '=== Generate key pair ==='
+    mkdir -p ${GIT_CHECKOUT_DIR}/jenkins/key
+    ssh-keygen -b 2048 -t rsa -f  "${GIT_CHECKOUT_DIR}/jenkins/key/antrea-ci-key" -q -N ""
+    publickey="$(cat ${GIT_CHECKOUT_DIR}/jenkins/key/antrea-ci-key.pub)"
+
+    echo "=== namespace value substitution ==="
+    mkdir -p ${GIT_CHECKOUT_DIR}/jenkins/out
+    cp ${GIT_CHECKOUT_DIR}/ci/cluster-api/vsphere/templates/* ${GIT_CHECKOUT_DIR}/jenkins/out
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAME/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s|SSHAUTHORIZEDKEYS|${publickey}|g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/namespace.yaml
+
+    echo "=== network spec value substitution==="
+    cluster_defaults=${WORK_HOME}/utils/CLUSTERDEFAULTS
+    while IFS= read -r line; do
+        IFS='=' read -ra kv <<< "$line"
+        sed -i "s|${kv[0]}|${kv[1]}|g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    done < "$cluster_defaults"
+
+    echo '=== Create a cluster in management cluster ==='
+    kubectl apply -f "${GIT_CHECKOUT_DIR}/jenkins/out/namespace.yaml"
+    kubectl apply -f "${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml"
+
+    echo '=== Wait for workload cluster secret for 10 min ==='
+    for t in {1..10}
+    do
+        sleep 1m
+        echo '=== Get kubeconfig (try for 1m) ==='
+        if kubectl get secret/${CLUSTER}-kubeconfig -n${CLUSTER} ; then
+            kubectl get secret/${CLUSTER}-kubeconfig -n${CLUSTER} -o json \
+            | jq -r .data.value \
+            | base64 --decode \
+            > "${GIT_CHECKOUT_DIR}/jenkins/out/kubeconfig"
+            SECRET_EXIST=true
+            break
+        fi
+    done
+
+    if [[ "$SECRET_EXIST" == false ]]; then
+        echo "=== Failed to get secret ==="
+        saveLogs
+        kubectl delete ns ${CLUSTER}
+        exit 1
+    else
+        export KUBECONFIG="${GIT_CHECKOUT_DIR}/jenkins/out/kubeconfig"
+        echo "=== Waiting all nodes up for 10 min ==="
```

Comment:
"Waiting for 10 minutes for all nodes to be up"

Reply:
Updated.